examples/example_app/example_app/jobs.py

```diff
@@ -3,23 +3,271 @@
 )
-from nautobot.dcim.models import Device, Location
+from nautobot.dcim.models import Device, Location, LocationType
 from nautobot.extras.choices import ObjectChangeActionChoices
 from nautobot.extras.jobs import get_task_logger
+from nautobot.extras.models import Status
 
-name = "Example App jobs"
+name = "Example App jobs"  # The "grouping" that will contain all Jobs defined in this file.
+
+
+class ExampleEverythingJob(Job):
+    """An example Job that uses and documents as much Job functionality as possible."""
+
+    class Meta:
+        """Metaclass attributes, describing and configuring this Job."""
+
+        name = "Example Job of Everything"
+        description = """\
+This is a Job that demonstrates as many Job features as it can.
+
+- All of the magic methods (before_start, run, on_success, on_failure, after_return).
+- All of the Meta attributes.
+- All of the ScriptVariable types as Job inputs.
+- Etc."""
+        approval_required = True  # Default: False
+        # Set to True to make it so that another user must approve any run of this Job
+        # before it can be executed.
+        dryrun_default = False  # Default: False
+        # Set to True to default Job execution to "dry-run" mode.
+        # Exactly what that means is up to the Job implementation.
+        field_order = []  # Default: unspecified (empty list)
+        # The order the input variables appear in the UI.
+        # If unspecified, they appear in declaration order.
+        has_sensitive_variables = False  # Default: True
+        # Set to False to assert that this Job's variables are non-sensitive;
+        # that is, they are permissible to save to the database and be made visible to
+        # any user who can view this Job's results.
+        hidden = False  # Default: False
+        # Set to True for a Job that's not shown in the UI by default.
+        read_only = False  # Default: False
+        # Set to True to assert that this Job does not make any changes to Nautobot data or the
+        # broader environment; exactly what this means is up to the job implementation.
+        soft_time_limit = 1000  # Default: None (follow global configuration)
+        # Time in seconds before Celery will automatically attempt to gracefully end the Job
+        # by raising a SoftTimeLimitExceeded exception.
+        task_queues = []  # Default: []
+        # The list of task queue names this Job wants to be routable to.
+        # Typically you'll configure this in the database, not hard-code it into the Job itself.
+        template_name = ""  # Default: ""
+        # App-only feature at present as only Apps can provide additional Django template files.
+        # Relative path of a Django template to display this Job's inputs in the UI for execution.
+        time_limit = 2000  # Default: None (follow global configuration)
+        # Time in seconds before Celery will automatically attempt terminate the Job by killing the
+        # Celery worker process.
```

Comment:
Let's add an `is_singleton` flag to the ExampleEverythingJob now that that's an option in 2.4.